src/framework/execution/__tests__/validation.test.ts

```diff
@@ -0,0 +1,78 @@
+import { IntegrationStep, IntegrationStepStartStates } from '../types';
+
+import { validateStepStartStates } from '../validation';
+
+describe('validateStepStartStates', () => {
+  test('throws error if step is provided in start states', () => {
+    const states: IntegrationStepStartStates = {
+      a: {
+        disabled: false,
+      },
+      b: {
+        disabled: true,
+      },
+    };
+    const steps: IntegrationStep[] = [
+      {
+        id: 'a',
+        name: 'a',
+        types: [],
+        executionHandler: jest.fn(),
+      },
+    ];
+
+    expect(() => validateStepStartStates(steps, states)).toThrow(
+      `Invalid step id "b" found in start states.`,
```

Comment:
```suggestion
      `Unknown steps found in start states: "b".`,
```
Should this handle more than one, as the following does, so that all are reported in a single error?

Reply:
Good suggestion, that would be a lot more friendly.